src/criteriaBuilder.js

```diff
@@ -0,0 +1,69 @@
+import extend from 'extend';
+
+export class CriteriaBuilder {
+
+  updateCriteria() {
+    let blocks = [];
+
+    this.filters.forEach((block, index) => {
+      let filtersParsed = {};
+
+      block.forEach(filter => {
+        if (!this.hasValue(filter.data.value)) {
+          return;
+        }
+
+        filtersParsed = extend(filtersParsed, this.parseField(filter.data.field, this.parseOperator(filter.data)));
+      });
+
+      blocks.push(filtersParsed);
+    });
+
+    this.criteria = (blocks.length > 1) ? {where: {or: blocks}} : {where: blocks[0]};
+  }
+
+  parseOperator(filter) {
+    switch(filter.operator) {
+      case 'between':
+        return this.parseBetween(filter);
+      case 'in':
+        return this.toArray(filter.value);
+      case 'not-in':
+        filter.operator = 'not';
+        filter.value    = this.toArray(filter.value);
+      default:
+        return {[filter.operator]: filter.value};
+    }
+  }
+
+  parseBetween(data, valueOperator = '>=', betweenOperator = '<=') {
```

Comment:
same here i guess. would be nice for other to have it configurable as the defaults are used above